app/main.py

```diff
@@ -1,2 +1,146 @@
+from __future__ import annotations
+from typing import Any, Hashable
+
+INITIAL_CAPACITY = 8
+LOAD_FACTOR = 2 / 3
+RESIZE_FACTOR = 2
+
+
+class Node:
+    def __init__(self, key: Hashable, value: Any) -> None:
+        self.key = key
+        self.value = value
+        self.hash_key = hash(key)
+
+
 class Dictionary:
-    pass
+
+    def __init__(self, **kwargs: Any) -> None:
+        self.length = 0
+        self.dict_keys: list = []
+        self.dict_values: list = []
+        self.capacity: int = INITIAL_CAPACITY
+        self.hash_table: list = [None] * self.capacity
+        for key, value in kwargs.items():
+            self.__setitem__(key, value)
+
+    def __str__(self) -> str:
+        result_str = "{"
+
+        for i in range(len(self.hash_table)):
+
+            if self.hash_table[i] is not None:
+                result_str += (f"'{self.hash_table[i].key}': "
+                               f"{self.hash_table[i].value}, ")
```

Comment:
You can simply iterate by non empty node, instead making range object for whole hash_table and iterating on table +few refactoring for more maintenance(instead reloading string whole time) and easier reading(in my opinion):
        items = []
        for node in self._hash_table:
            if node is not None:
                items.append(f"'{node.key}': {node.value}")
        return '{' + ', '.join(items) + '}'

Reply:
Yep, it's simplier.
Thank's.